app/main.py

```diff
@@ -2,0 +1,45 @@
+from __future__ import annotations
+
+
+class Animal:
+    alive: list[Animal] = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        self.add_to_alive(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+    @classmethod
+    def add_to_alive(cls, animal: Animal) -> None:
+        cls.alive.append(animal)
+
+    @classmethod
+    def remove_from_alive(cls, animal: Animal) -> None:
+        cls.alive.remove(animal)
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if self.hidden:
+            self.hidden = False
+        else:
+            self.hidden = True
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Herbivore) and herbivore.hidden is False:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                Animal.remove_from_alive(herbivore)
```

Comment:
The `bite` method should check if the `herbivore` is still in the `Animal.alive` list before attempting to bite. This ensures that only living herbivores can be bitten.